File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/common/ActorState.java
package org.pyjjs.scheduler.core.api.impl.actors.common;

import akka.actor.ActorContext;
import akka.actor.ActorRef;
import akka.actor.ActorSystem;

public abstract class ActorState {

    private ActorContext actorContext;

    private boolean initialized = false;

    public ActorState(ActorContext actorContext) {
        this.actorContext = actorContext;
    }

    public ActorRef getActorRef() {
        return actorContext.self();
    }

    public ActorSystem getActorSystem() {
        return actorContext.system();
    }

    public ActorContext getActorContext() {
        return actorContext;
    }

    public boolean isInitialized() {
        return initialized;
    }

    public void setInitialized(boolean initialized) {
        this.initialized = initialized;
    }

    protected abstract ActorState copySelf();

}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/common/ActorState.kt
package org.pyjjs.scheduler.core.api.impl.actors.common

import akka.actor.ActorContext
import akka.actor.ActorRef
import akka.actor.ActorSystem

abstract class ActorState(val actorContext: ActorContext) {

    var isInitialized = false

    val actorRef: ActorRef
        get() = actorContext.self()

    val actorSystem: ActorSystem
        get() = actorContext.system()

    abstract fun copySelf(): ActorState

}
